src/legacy/ui/public/chrome/directives/header_global_nav/components/header.tsx

```diff
@@ -88,11 +91,11 @@ const TRUNCATE_LIMIT: number = 64;
 const TRUNCATE_AT: number = 58;
 
 function extendRecentlyAccessedHistoryItem(
-  navLinks: NavLink[],
+  navLinks: ChromeNavLink[],
   recentlyAccessed: RecentlyAccessedHistoryItem
 ) {
-  const href = relativeToAbsolute(chrome.addBasePath(recentlyAccessed.link));
-  const navLink = navLinks.find(nl => href.startsWith(nl.subUrlBase));
+  const href = chrome.addBasePath(recentlyAccessed.link);
+  const navLink = navLinks.find(nl => href.startsWith(nl.appUrl));
```

Comment:
This no longer finds the navLink for recently accessed items when basePath is in use, because `nl.appUrl` doesn't have the basePath, but `href` does.